test/run_error_tests.py

```
@@ -0,0 +1,124 @@
+
+class MockFileSystem:
+    def __init__(self):
+        self.files = {}
+        self.access_errors = False
+        
+    def set_access_error(self, state: bool):
+        self.access_errors = state
+        
+    def read_file(self, path: str) -> str:
+        if self.access_errors:
+            raise AccessError(f"Cannot access {path}")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)